arelle/plugin/validate/ESEF/Dimensions.py

```diff
@@ -18,28 +19,40 @@
 try:
     import regex as re
 except ImportError:
-    import re
+    import re  # type: ignore[no-redef]
+from collections.abc import Callable
+from arelle.ValidateXbrl import ValidateXbrl
+from arelle.ModelDtsObject import ModelConcept
+
+_: Callable[[str], str]  # Handle gettext
 
-def checkFilingDimensions(val):
+
+def checkFilingDimensions(val: ValidateXbrl) -> None:
 
     val.primaryItems = set() # concepts which are line items (should not also be dimension members
-    val.domainMembers = set() # concepts which are dimension domain members
+    val.domainMembers = set()  # concepts which are dimension domain members
+
     elrPrimaryItems = defaultdict(set)
-    hcPrimaryItems = set()
-    hcMembers = set()
-    
-    def addDomMbrs(sourceDomMbr, ELR, membersSet):
+    hcPrimaryItems: set[ModelConcept] = set()
+    hcMembers: set[Any] = set()
```

Comment:
Not in the world of mypy if I'm not misunderstanding you here. Setting `hcMembers = set()` yields this:
```
(.venv) vscode ➜ /workspaces/Arelle (add-type-hint-esef-dimensions ✗) $ mypy arelle --namespace-packages
arelle/plugin/validate/ESEF/Dimensions.py:38: error: Need type annotation for "hcMembers" (hint: "hcMembers: Set[<type>] = ...")  [var-annotated]
Found 1 error in 1 file (checked 263 source files)
```

It probably should be something other than `Any` but at the moment, I don't know exactly what. My thinking is that it could be adressed further down the line.

Reply:
Gotcha, it wants an explicit `Any`. Makes sense.